test/common/selfmanaged_apicast.go

```diff
@@ -0,0 +1,889 @@
+// This test case should prove that it is possible for customers to deploy self-managed APIcast
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	appsv1alpha1 "github.com/3scale/apicast-operator/pkg/apis/apps/v1alpha1"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/controllers/subscription/rhmiConfigs"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	routev1 "github.com/openshift/api/route/v1"
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"net/http"
+	"os"
+	"os/exec"
+	ctrl "sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	apicastOperatorVersion       = "0.5.0"
+	apicastImageStreamTag        = "3scale2.11.0"
+	apicastNamespace             = "selfmanaged-apicast"
+	apiExampleApicast            = "apicast-example-apicast"
+	adminPortalCredentialsSecret = "adminportal-credentials"
+	namespacePrefix              = "redhat-rhoam-"
+	//defaultDedicatedAdminName    = "customer-admin"
+	accountOrgName               = "Developer"
+	serviceSystemName            = "api"
+)
+
+var (
+	log                 = l.NewLogger()
+	threeScaleNamespace string
+)
+
+func TestSelfmanagedApicast(t TestingTB, testingCtx *TestingContext) {
+	threeScaleNamespace = fmt.Sprintf("%s3scale", namespacePrefix)
+	scheme := runtime.NewScheme()
+	utilruntime.Must(operatorsv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(v1.AddToScheme(scheme))
+	utilruntime.Must(routev1.AddToScheme(scheme))
+	utilruntime.Must(appsv1alpha1.SchemeBuilder.AddToScheme(scheme))
+
+	config := ctrl.GetConfigOrDie()
+	client, err := k8sclient.New(config, k8sclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		t.Fatalf("Error get Client k8sclient: %v", err)
+	}
+	ctx := context.Background()
+
+	err = cleanUpBeforeTest(ctx, client)
+	if err != nil {
+		t.Fatalf("Error clean before test: %v", err)
+	}
+
+	// Create apicast namespace
+	err = createApicastNamespace(ctx, client)
+	if err != nil {
+		t.Fatalf("Error create namespace  %s: %v", apicastNamespace, err)
+	}
+
+	// Get 3scale admin token
+	token3scale, err := getThreeScaleAdminToken(ctx, client)
+	if err != nil {
+		t.Fatalf("Error get 3scale admin Token: %v", err)
+	}
+
+	// Customer login
+	err = customerLogin()
+	if err != nil {
+		t.Fatalf("Error in customer login: %v", err)
+	}
```

Comment:
Wondering have you seen this error before even with the IDP already set up ?
```
time="2022-01-06T11:21:40Z" level=info msg="Creating namespace selfmanaged-apicast"
Now using project "selfmanaged-apicast" on server "https://api.chfan.je0y.s1.devshift.org:6443".
time="2022-01-06T11:21:45Z" level=info msg="Get 3scale admin Token"
time="2022-01-06T11:21:45Z" level=info msg="Customer Login"
Login failed (401 Unauthorized)
Verify you have provided correct credentials.
time="2022-01-06T11:21:46Z" level=error msg="Error: " error="exit status 1"
time="2022-01-06T11:21:46Z" level=error msg="Error login customer: " error="exit status 1"
```
Something similar happened in a pipeline once and found that scaling the `authentication-operator` in `openshift-authentication-operator` helped. And retrying to login over a duration of time :thinking: 
* https://gitlab.cee.redhat.com/integreatly-qe/ci-cd/-/blob/master/pipelines/managed-api-performance-test.groovy#L255-267
```
oc scale deployment authentication-operator -n openshift-authentication-operator --replicas=0
oc scale deployment authentication-operator -n openshift-authentication-operator --replicas=1
```

Reply:
Error in customer login...  Yes I saw it. I checked manually, from command line - "oc login -u customer-admin01 -p Password1". It didn't work. Issue was with IDP. After IDP reinstalling and check that manually it's working - script also working. I saw also that after IDP installation, and even it present in OCM UI - it still missing, if trying "oc get identity". There are few different ways i used to see "identity" in command line. After it present, seems all is working, but better seems just check "oc login -u customer-admin01 -p Password1"